### PR TITLE
Atom/mriegger/decalnormalmaps

### PR DESCRIPTION
**Pull request by mrieggeramzn**
Adding initial normal map support to the decals.
These get back into a separate texture array. (This is because they use different formats, BC7 for color and BC5 for normals)

Tested Editor and ASV

debug1
debug2
debug3

### CODE REVIEW

**Review by amzn-tommy — APPROVED** [review]
This seems like a good case for an update to the AtomSampleViewer integration tests. If you haven't already, in addition to the examples in the screenshots you posted make sure you test with multiple decals overlapping and with the decal projecting onto something more interesting than the ground plane.

File: Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp [hunk 1]
```diff
@@ -290,6 +312,24 @@ namespace AZ
             return id.IsValid() && materialData.m_materialAssetData.IsReady();
         }
 
+        bool DecalTextureArray::AreAllTextureMapsPresent(const DecalMapType mapType) const
+        {
+            int iter = m_materials.begin();
+            while (iter != -1)
+            {
+                if (!IsTextureMapPresentInMaterial(m_materials[iter], mapType))
+                    return false;
```
[the remark this answers is not in the document]
Comment (mrieggeramzn) [reply Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp:321]: Good point. Ill be committing updated ASV decals soon and will add that.

File: Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp [hunk 2]
```diff
@@ -122,42 +125,42 @@ namespace AZ
...
         {
-            return m_textureArrayPacked;
+            return m_textureArrayPacked[mapType];
         }
 
         bool DecalTextureArray::IsValidDecalMaterial(const AZ::RPI::MaterialAsset& materialAsset)
         {
-            return GetStreamingImageAsset(materialAsset, AZ::Name(BaseColorTextureMapName)).IsReady();
+            return GetStreamingImageAsset(materialAsset, GetMapName(DecalMapType_Diffuse)).IsReady();
         }
```
Comment (antonmic) [inline Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp:146]: What if I want a decal that only changes the normals and doesn't affect the color?

Comment (antonmic) [reply Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArray.cpp:146]: As we discussed, if this is to be addressed in follow up work than I'm fine with this going in as is.

**Review by antonmic — APPROVED** [review]

**Pull request merged**